### DISCUSSION TITLE
Query regarding build infrastructure and continuous build validation

### DISCUSSION THREAD

**Discussion by vrukesh** [picolibc/picolibc#783] [General]
Hi All,

1. Do we have host any publicly visible results for build infrastructure and continuous build validation?

2. Do we execute any specific builds for picolibc with:
a. static analyzers (clang scan-view, cppcheck etc)
b. code coverage
c. sanitizers (UBSAN etc.)

Thanks and regards,
Vrukesh

**Comment by keith-packard** [comment]
Yup, all the CI is run here on github from scripts found in the .github directory in the source repository. We're currently using -fanalyzer with GCC for static analysis. Hooking up other analyzers is tricky because of the build environment, but I'd love to figure out how to add more checking.

All of the github CI data are available in the Actions tab, and the current CI status for the 'main' branch are presented on the github front page.

**Reply by vrukesh** [reply]
Hi Keith,

Thanks for detailed reply ! 

We observed some build issues with recent changes in picolibc compiled with LLVM clang. (You might possibly well aware of bare-metal LLVM based toolchain for ARM + picolibc: https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/tree/main, which is used for this build)

We see two configs which use clang compiler with "set(CMAKE_C_COMPILER clang)".
https://github.com/picolibc/picolibc/blob/main/cmake/TC-clang-thumbv6m.cmake
https://github.com/picolibc/picolibc/blob/main/cmake/TC-clang-thumbv7m.cmake

1. Could you kindly help to point to build bot for these cmake?
2. Wondering if any specific build bots use LLVM clang to build+test picolibc?

Thanks and Regards,
Vrukesh

**Reply by keith-packard** [reply]
Cmake is really only designed to be used by Zephyr; if you're building a complete toolchain, you'll really want to use the meson bits instead. I should probably stick more dire warning messages for people who think the cmake stuff is going to give them good results...
